waspc/src/Wasp/Generator/WebAppGenerator/AuthG.hs

```diff
@@ -50,5 +55,71 @@ genCreateAuthRequiredPage auth =
 genUseAuth :: Generator FileDraft
 genUseAuth = return $ C.mkTmplFd (C.asTmplFile [relfile|src/auth/useAuth.js|])
 
+genAuthForms :: AS.Auth.Auth -> Generator [FileDraft]
+genAuthForms auth =
+  sequence
+    [ genLoginForm auth,
+      genSignupForm auth,
+      genAuthForm auth,
+      genStitchesConfig,
+      genSocialIcons
+    ]
+
+genSocialIcons :: Generator FileDraft
+genSocialIcons =
+    copyTmplFile [relfile|auth/forms/SocialIcons.jsx|]
+  where
+    copyTmplFile = return . C.mkSrcTmplFd
+
+genStitchesConfig :: Generator FileDraft
+genStitchesConfig =
+    -- TODO(matija): where should I put this? Maybe keep it localy scopped in auth/forms for now?
+    copyTmplFile [relfile|stitches.config.js|]
+  where
+    -- TODO(matija): duplicated, extract
+    copyTmplFile = return . C.mkSrcTmplFd
+
+genAuthForm :: AS.Auth.Auth -> Generator FileDraft
+genAuthForm auth =
+  compileTmplToSamePath
+    [relfile|auth/forms/Auth.jsx|]
+    [ "onAuthSucceededRedirectTo" .= getOnAuthSucceededRedirectToOrDefault auth
+    , "isUsernameAndPasswordAuthEnabled" .= AS.Auth.isUsernameAndPasswordAuthEnabled auth
+    , "isBothExternalAndUsernameAndPasswordAuthEnabled" .= AS.Auth.isBothExternalAndUsernameAndPasswordAuthEnabled auth
+       -- Social auth methods
+    , "isExternalAuthEnabled" .= AS.Auth.isExternalAuthEnabled auth
+    -- Google
+    , "isGoogleAuthEnabled" .= AS.Auth.isGoogleAuthEnabled auth
+    , "googleSignInPath" .= OAuth.serverLoginUrl googleAuthProvider
+    -- GitHub
+    , "isGitHubAuthEnabled" .= AS.Auth.isGitHubAuthEnabled auth
+    , "gitHubSignInPath" .= OAuth.serverLoginUrl gitHubAuthProvider
+    ]
+
+genLoginForm :: AS.Auth.Auth -> Generator FileDraft
+genLoginForm auth =
+  compileTmplToSamePath
+    [relfile|auth/forms/Login.jsx|]
+    [ "onAuthSucceededRedirectTo" .= getOnAuthSucceededRedirectToOrDefault auth
+    ]
+
+genSignupForm :: AS.Auth.Auth -> Generator FileDraft
+genSignupForm auth =
+  compileTmplToSamePath
+    [relfile|auth/forms/Signup.jsx|]
+    [ "onAuthSucceededRedirectTo" .= getOnAuthSucceededRedirectToOrDefault auth
+    ]
+
+compileTmplToSamePath :: Path' Rel' File' -> [Pair] -> Generator FileDraft
```

Comment:
I'm not sure if you need this function, there is already `mkTmplFdWithData` in `Wasp.Generator.WebAppGenerator.Common`

Reply:
Simplified helper function with `mkTmplFdWithData`